web/src/LogPane.tsx

```diff
@@ -53,6 +53,52 @@ class LogPane extends Component<LogPaneProps, LogPaneState> {
     }
   }
 
+  componentDidUpdate(prevProps: LogPaneProps) {
+    if (this.props.highlightsEnabled && this.props.modalIsOpen) {
+      document.removeEventListener(
```

Comment:
in general, it's going to be pretty racey/error-prone to add/remove listeners in componentDidUpdate.

have you considered adding these checks to handleSelectionChange, and simply doing nothing in that case?

Reply:
Good point! This simplifies the code nicely.